### PR TITLE
Update MapColor field names

### LINKED ISSUE #1451: Material-/MapColor Field Renames
For longterm discussion about how to best rename them. 

It should be noted that while bad, they are at least moderately functional (when you know they're for colors on a map).

The field names are currently named using three different systems:
- The color that it actually is
- The material that uses the color (outdated/abbreviated names)
- The block that uses the color

At the very least they should all follow a unified naming scheme, of which there are three practical ones:
- Naming color based on the color it actually is
    - Bikeshed time
    - Difficult as we have about 12 shades of brown.
- Based on the blocks that use it
    - More work to compile, maintain, and leads to names similar to the new Material one's, but with more mess as color is _per blockstate_, so this is likely impossible to do well.
- The hex value of the color
    -  Each rename is simple, but now each name is also useless unless you can interpret hex color codes.

The first option would be the most reasonable to use. We can also "cheat" a little and refer to MC textures, though that will require a Javadoc disclaimer (eg Emerald Green doesn't mean dark green, but MC's emerald color). Slightly complicated by MC's naming being off for some colors. Naming them in terms of *blockReference_baseColor* would make things more uniform and easier to understand, but would mean 16 more renames for the terracotta colors.

Here are all the color's hex values, and setup so Idea will show you the color on the side: https://pastebin.com/nTXgt3Dr

### PR DESCRIPTION
**Pull request by dexman545**
Resolves #1451

### CODE REVIEW

File: mappings/net/minecraft/block/MapColor.mapping [hunk 1]
```diff
@@ -1,60 +1,69 @@
 CLASS net/minecraft/class_3620 net/minecraft/block/MapColor
 	COMMENT Represents the surface color of a block when rendered from the {@link net.minecraft.client.gui.MapRenderer}.
-	FIELD field_15976 CLAY Lnet/minecraft/class_3620;
+	COMMENT Color names referencing a material or object refer to their vanilla MineCraft textures, not their real-world counterparts, eg. "emerald green."
```
Comment (i509VCB) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	COMMENT Color names refer to a material or an object which refers to their vanilla Minecraft textures, not their real-world counterparts, eg. "emerald green."
```

Even after my suggestion I do believe this needs another look from someone else.

File: mappings/net/minecraft/block/MapColor.mapping [hunk 2]
```diff
@@ -1,60 +1,69 @@
 CLASS net/minecraft/class_3620 net/minecraft/block/MapColor
 	COMMENT Represents the surface color of a block when rendered from the {@link net.minecraft.client.gui.MapRenderer}.
-	FIELD field_15976 CLAY Lnet/minecraft/class_3620;
+	COMMENT Color names refer to a material or an object which refers to their vanilla Minecraft textures, not their real-world counterparts, eg. "emerald green."
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	COMMENT Color names refer to a material or an object which refers to their vanilla Minecraft textures, not their real-world counterparts, eg. "emerald green".
```

File: mappings/net/minecraft/block/MapColor.mapping [hunk 3]
```diff
@@ -1,60 +1,69 @@
...
 	FIELD field_15978 GRAY Lnet/minecraft/class_3620;
-	FIELD field_15979 WEB Lnet/minecraft/class_3620;
-	FIELD field_15980 LAPIS Lnet/minecraft/class_3620;
-	FIELD field_15981 ORANGE_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15982 RED_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15983 DIAMOND Lnet/minecraft/class_3620;
+	FIELD field_15979 WHITE_GRAY Lnet/minecraft/class_3620;
+	FIELD field_15980 LAPIS_BLUE Lnet/minecraft/class_3620;
+	FIELD field_15981 TERRACOTTA_ORANGE Lnet/minecraft/class_3620;
+	FIELD field_15982 TERRACOTA_RED Lnet/minecraft/class_3620;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	FIELD field_15982 TERRACOTTA_RED Lnet/minecraft/class_3620;
```

File: mappings/net/minecraft/block/MapColor.mapping [hunk 4]
```diff
@@ -1,60 +1,69 @@
...
-	FIELD field_15988 LIGHT_GRAY_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15989 PINK_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15990 CYAN_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15991 LIGHT_BLUE_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_15992 BROWN_TERRACOTTA Lnet/minecraft/class_3620;
+	FIELD field_15988 TERRACOTA_LIGHT_GRAY Lnet/minecraft/class_3620;
+	FIELD field_15989 TERRACOTTA_PINK Lnet/minecraft/class_3620;
+	FIELD field_15990 TERRACOTA_CYAN Lnet/minecraft/class_3620;
+	FIELD field_15991 TERRACOTTA_LIGHT_BLUE Lnet/minecraft/class_3620;
+	FIELD field_15992 TERRACOTA_BROWN Lnet/minecraft/class_3620;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	FIELD field_15988 TERRACOTTA_LIGHT_GRAY Lnet/minecraft/class_3620;
	FIELD field_15989 TERRACOTTA_PINK Lnet/minecraft/class_3620;
	FIELD field_15990 TERRACOTTA_CYAN Lnet/minecraft/class_3620;
	FIELD field_15991 TERRACOTTA_LIGHT_BLUE Lnet/minecraft/class_3620;
	FIELD field_15992 TERRACOTTA_BROWN Lnet/minecraft/class_3620;
```

File: mappings/net/minecraft/block/MapColor.mapping [hunk 5]
```diff
@@ -1,60 +1,69 @@
...
+	FIELD field_15999 PALE_GREEN Lnet/minecraft/class_3620;
+	FIELD field_16000 DIRT_BROWN Lnet/minecraft/class_3620;
+	FIELD field_16001 EMERALD_GREEN Lnet/minecraft/class_3620;
+	FIELD field_16002 BRIGHT_RED Lnet/minecraft/class_3620;
+	FIELD field_16003 TERRACOTTA_WHITE Lnet/minecraft/class_3620;
+	FIELD field_16004 DARK_GREEN Lnet/minecraft/class_3620;
+	FIELD field_16005 IRON_GRAY Lnet/minecraft/class_3620;
 	FIELD field_16006 COLORS [Lnet/minecraft/class_3620;
-	FIELD field_16007 BLACK_TERRACOTTA Lnet/minecraft/class_3620;
+	FIELD field_16007 TERRACOTA_BLACK Lnet/minecraft/class_3620;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	FIELD field_16007 TERRACOTTA_BLACK Lnet/minecraft/class_3620;
```

File: mappings/net/minecraft/block/MapColor.mapping [hunk 6]
```diff
@@ -1,60 +1,69 @@
...
-	FIELD field_16013 YELLOW_TERRACOTTA Lnet/minecraft/class_3620;
+	FIELD field_16012 DARK_RED Lnet/minecraft/class_3620;
+	FIELD field_16013 TERRACOTTA_YELLOW Lnet/minecraft/class_3620;
 	FIELD field_16014 PURPLE Lnet/minecraft/class_3620;
-	FIELD field_16015 BLUE_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_16016 ICE Lnet/minecraft/class_3620;
-	FIELD field_16017 SPRUCE Lnet/minecraft/class_3620;
-	FIELD field_16018 LIME_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_16019 WATER Lnet/minecraft/class_3620;
+	FIELD field_16015 TERRACOTA_BLUE Lnet/minecraft/class_3620;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	FIELD field_16015 TERRACOTTA_BLUE Lnet/minecraft/class_3620;
```

File: mappings/net/minecraft/block/MapColor.mapping [hunk 7]
```diff
@@ -1,60 +1,69 @@
...
 	FIELD field_16024 LIGHT_BLUE Lnet/minecraft/class_3620;
-	FIELD field_16025 QUARTZ Lnet/minecraft/class_3620;
+	FIELD field_16025 OFF_WHITE Lnet/minecraft/class_3620;
 	FIELD field_16026 CYAN Lnet/minecraft/class_3620;
-	FIELD field_16027 GRAY_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_16028 GREEN_TERRACOTTA Lnet/minecraft/class_3620;
-	FIELD field_16029 PURPLE_TERRACOTTA Lnet/minecraft/class_3620;
+	FIELD field_16027 TERRACOTTA_GRAY Lnet/minecraft/class_3620;
+	FIELD field_16028 TERRACOTA_GREEN Lnet/minecraft/class_3620;
+	FIELD field_16029 TERRACOTA_PURPLE Lnet/minecraft/class_3620;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/MapColor.mapping]:
```suggestion
	FIELD field_16028 TERRACOTTA_GREEN Lnet/minecraft/class_3620;
	FIELD field_16029 TERRACOTTA_PURPLE Lnet/minecraft/class_3620;
```

**Review by YanisBft — COMMENTED** [review]
Also mind resolve conflicts

**Comment by dexman545** [comment]
@YanisBft fixed

**Review by YanisBft — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

**Comment by dexman545** [comment]
Retargeted

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/block/MapColor.mapping [hunk 8]
```diff
@@ -1,67 +1,69 @@
 CLASS net/minecraft/class_3620 net/minecraft/block/MapColor
 	COMMENT Represents the surface color of a block when rendered from the {@link net.minecraft.client.gui.MapRenderer}.
-	FIELD field_15976 CLAY Lnet/minecraft/class_3620;
+	COMMENT Color names refer to a material or an object which refers to their vanilla Minecraft textures, not their real-world counterparts, eg. "emerald green".
+	COMMENT Names are in the form of <i>blockReference_baseColor</i>.
```
Comment (Juuxel) [inline mappings/net/minecraft/block/MapColor.mapping]: Slightly incorrect as this doesn't apply to all of them. (for example, `BROWN` or `OFF_WHITE`)

File: mappings/net/minecraft/block/MapColor.mapping [hunk 9]
```diff
@@ -1,67 +1,69 @@
...
+	FIELD field_16015 TERRACOTTA_BLUE Lnet/minecraft/class_3620;
+	FIELD field_16016 PALE_PURPLE Lnet/minecraft/class_3620;
+	FIELD field_16017 SPRUCE_BROWN Lnet/minecraft/class_3620;
+	FIELD field_16018 TERRACOTTA_LIME Lnet/minecraft/class_3620;
+	FIELD field_16019 WATER_BLUE Lnet/minecraft/class_3620;
 	FIELD field_16020 RED Lnet/minecraft/class_3620;
 	FIELD field_16021 id I
 	FIELD field_16022 WHITE Lnet/minecraft/class_3620;
-	FIELD field_16023 STONE Lnet/minecraft/class_3620;
+	FIELD field_16023 DIM_GRAY Lnet/minecraft/class_3620;
```
Comment (Juuxel) [inline mappings/net/minecraft/block/MapColor.mapping]: What about just `STONE_GRAY`? That's a bit more descriptive.

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/MapColor.mapping (hunk 8 above)
Comment (Juuxel) [reply mappings/net/minecraft/block/MapColor.mapping]: Uh, why was this resolved? It's still not true as all names are not in that form.

Comment (dexman545) [reply mappings/net/minecraft/block/MapColor.mapping]: Should be clearer now

**Review by Juuxel — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Comment by dexman545** [comment]
Retargeted

**Pull request merged**